src/utils/validation.ts

```diff
@@ -17,14 +17,14 @@ const validate = <T extends IConfigSchema, U extends { [key: string]: string }>(
   obj: U
 ): void => {
   if (schema) {
-  }
-  const errors = Object.keys(schema)
-    .filter(key => !schema[key](obj[key]))
-    .map(key => new Error(`${key} is invalid.`));
+    const errors = Object.keys(schema)
+      .filter(key => !schema[key](obj[key]))
+      .map(key => new Error(`${key} is invalid.`));
 
-  if (errors.length > 0) {
-    for (const { message } of errors) {
-      throw new Error(message);
+    if (errors.length > 0) {
+      for (const { message } of errors) {
+        throw new Error(message);
```

Comment:
why do we create a new `Error` here? as I understand `errors` is already a list of `Error` object that you just created.
maybe you could write something like:
```
for (const error of errors) {
      throw new Error(message);
```
but I think it would make sense to just select the first error and throw it instead of using a loop with an immediate exit after the first iteration.
```
const [error] = errors
throw error;
```